SimulationAgent/Simulation/DeviceActor.cs

```diff
@@ -178,21 +154,24 @@ public class DeviceActor : IDeviceActor
 
         public DeviceActor(
             ILogger logger,
-            DependencyResolution.IFactory factory)
+            Connect connectLogic,
+            DeviceBootstrap deviceBootstrapLogic,
+            UpdateDeviceState updateDeviceStateLogic,
+            UpdateReportedProperties updateReportedPropertiesLogic,
+            SendTelemetry sendTelemetryLogic,
+            ITimer cancellationCheckTimer)
         {
             this.log = logger;
-            this.deviceBootstrapLogic = factory.Resolve<DeviceBootstrap>();
-            this.connectLogic = factory.Resolve<Connect>();
-            this.updateDeviceStateLogic = factory.Resolve<UpdateDeviceState>();
-            this.sendTelemetryLogic = factory.Resolve<SendTelemetry>();
-            this.updateReportedPropertiesLogic = factory.Resolve<UpdateReportedProperties>();
-            this.factory = factory;
+
+            this.connectLogic = connectLogic;
```

Comment:
just a matter of preference, I try to avoid using the DI factory when the DI framework can take care of building instances for me. For unit tests, and to get a feel about the local complexity, I find it easier to see the dependencies explicitly listed in the ctor.